django/common/kafka/producer.py

```diff
@@ -17,11 +18,13 @@ def default(self, obj):
             return obj.isoformat()
         if isinstance(obj, bytes):
             # FIXME: this should be done differently, issue #376
-            # Python doesn't know how to serialize bytes
-            # We arbitrarily choose utf-8. Note that if we need to decode bytes another
+            # Python doesn't know how to serialize bytes so we use chardet to find the
+            # right encoding.
+            # Note that if we need to decode bytes another
             # way, we'll need to use a cleaning script that basically does
-            # val.encode("utf-8").decode("<your-encoding>")
-            return obj.decode("utf-8")
+            # val.encode("<detected-encoding>").decode("<your-encoding>")
+            detected_encoding = chardet.detect(obj)["encoding"]
```

Comment:
lol so powerful.
doesn't this raise a KeyError if `chardet` fails to detect the encoding ?

Reply:
I think `detect` always choose some encoding, even if it has 0 confidence in it